include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1126,7 +1151,9 @@ __parallel_find_or(oneapi::dpl::__internal::__device_backend_tag, _ExecutionPoli
         oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<__find_or_kernel, _CustomName, _Brick,
                                                                                _BrickTag, _Ranges...>;
 
-    constexpr bool __or_tag_check = ::std::is_same_v<_BrickTag, __parallel_or_tag>;
+    assert("This device does not support 64-bit atomics" &&
```

Comment:
This is a bigger question than this PR is looking to address, but should we have a "general" version of this algorithm which supports hardware which doesn't support 64-bit atomics rather than merely an assert? Obviously this would come at a cost, and additional code.

For instance, I believe this is not supported in FPGA (if I remember correctly from work on histogram), which also may use 64-bit atomics based on the types passed in from the user).

Note that this is not a new problem added by this PR or any recent work, but just one which is been brought up by the addition of the assertion.

Reply:
Probably we have one more option here: to replace this `assert` to `static_assert` and check `sizeof(_Type)`.
If problem really exist I think that there is no reasons to land this PR due it's about only 10% perf profit.

Reply:
@akukanov what do you think about this? I mean the case described by @danhoeflinger at https://github.com/oneapi-src/oneDPL/pull/1650#discussion_r1658876487 ?

Reply:
Probably this point has been already resolved early: in the structures `__parallel_find_forward_tag` and `__parallel_find_backward_tag` the `_AtomicType` has been defined as below:
```C++
// FPGA devices don't support 64-bit atomics
#if _ONEDPL_FPGA_DEVICE
    using _AtomicType = uint32_t;
#else
    using _AtomicType = oneapi::dpl::__internal::__difference_t<_RangeType>;
#endif
```